backend/testfiles/execution/stdlib/semanticTokenization.dark

```diff
@@ -871,6 +871,19 @@ module TokenizeExpression =
                                 (TokenType.String, (0L, 1L), (0L, 6L))
                                 (TokenType.Symbol, (0L, 6L), (0L, 7L)) ]
 
+  ("$\"hello {name}\"" |> tokenize) = [ (TokenType.Symbol, (0L, 0L), (0L, 1L))
+                                        (TokenType.Symbol, (0L, 1L), (0L, 2L))
+                                        (TokenType.String, (0L, 2L), (0L, 3L))
+                                        (TokenType.String, (0L, 3L), (0L, 4L))
+                                        (TokenType.String, (0L, 4L), (0L, 5L))
+                                        (TokenType.String, (0L, 5L), (0L, 6L))
+                                        (TokenType.String, (0L, 6L), (0L, 7L))
+                                        (TokenType.String, (0L, 7L), (0L, 8L))
+                                        (TokenType.Symbol, (0L, 8L), (0L, 9L))
+                                        (TokenType.VariableName, (0L, 9L), (0L, 13L))
+                                        (TokenType.Symbol, (0L, 13L), (0L, 14L))
+                                        (TokenType.Symbol, (0L, 14L), (0L, 15L)) ]
```

Comment:
```suggestion
  ("$\"hello {name}\"" |> tokenize) = [ (TokenType.Symbol, (0L, 0L), (0L, 1L)) // $
                                        (TokenType.Symbol, (0L, 1L), (0L, 2L)) // "
                                        (TokenType.String, (0L, 2L), (0L, 3L)) // h
                                        (TokenType.String, (0L, 3L), (0L, 4L)) // e
                                        (TokenType.String, (0L, 4L), (0L, 5L)) // l
                                        (TokenType.String, (0L, 5L), (0L, 6L)) // l
                                        (TokenType.String, (0L, 6L), (0L, 7L)) // o
                                        (TokenType.String, (0L, 7L), (0L, 8L))
                                        (TokenType.Symbol, (0L, 8L), (0L, 9L))
                                        (TokenType.VariableName, (0L, 9L), (0L, 13L))
                                        (TokenType.Symbol, (0L, 13L), (0L, 14L))
                                        (TokenType.Symbol, (0L, 14L), (0L, 15L)) ]
```

This looks wrong - I made a few comments, but this breaks the tokenization _many_ individual tokens, rather than capturing `hello ` all at once.

Reply:
It makes me wonder if this string is being parsed with a _bunch_ of `StringSegment`s

Reply:
Not `StringSegment, it is capturing them as `string_text`
```
    string_interpolation_content: $ =>
      repeat1(choice($.string_to_eval, $.string_text)),

    string_text: $ =>
      prec.left(
        repeat1(
          choice(
            token.immediate(prec(1, /[^\\"{}\n]/)),
            $.char_or_string_escape_sequence,
          ),
        ),
      ),
```

couldn't really find a way to capture it as the full thing and at the same time keep the order of `string_text` and `string_to_eval` correct

Reply:
Hmm, I think it's worth figuring out what's going on here. If you want to merge and circle back to this, that's okay, but at some point we'll really want this to be one token. Imagine we have a _really long_ string, that involves a tiny bit of interpolation - that suddenly becomes a _lot_ of tokens. Plus, I think "being a single token" will influence your interactions in the editor - let's say I'm at `h|ello` and hit `ctrl+right` intending to jump to `hello|` - as-is, this might just jump to `he|llo`

Reply:
I agree, and I am not happy with this solution either, I'd prefer to look more into it before merging this.